src/ape/api/providers.py

```diff
@@ -60,17 +60,40 @@ class BlockAPI(BaseInterfaceModel):
     # NOTE: All fields in this class (and it's subclasses) should not be `Optional`
     #       except the edge cases noted below
 
+    """
+    The number of transactions in the block.
+    """
     num_transactions: int = 0
+
+    """
+    The block hash identifier.
+    """
     hash: Optional[Any] = None  # NOTE: pending block does not have a hash
+
+    """
+    The block number identifier.
+    """
     number: Optional[int] = None  # NOTE: pending block does not have a number
+
+    """
+    The preceeding block's hash.
+    """
     parent_hash: Any = Field(
         EMPTY_BYTES32, alias="parentHash"
     )  # NOTE: genesis block has no parent hash
-    size: int
+
+    """
+    The timestamp the block was produced.
+    """
     timestamp: int
 
+    _size: Optional[int] = None
```

Comment:
Think this makes sense to do? Might replace some of the logic used to do pop and stuff
```suggestion
    _size: Optional[int] = Field(None, alias="size")
```

Reply:
I don't think that would work, as the sub-class has a computed field of the same name.

Reply:
You actually get another problem anyway:

```
NameError: Fields must not use names with leading underscores; e.g., use 'size' instead of '_size'.
```

As mentioned on the PR for trying to dundarize some fields in events, **you cannot have fields that start with underscores in Pydantic v2**. So this is a no-go.